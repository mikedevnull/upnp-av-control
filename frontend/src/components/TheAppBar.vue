<template>
  <header class="mdc-top-app-bar">
    <div class="mdc-top-app-bar__row">
      <section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start">
        <span class="mdc-top-app-bar__title">UPnP AV controller</span>
      </section>
      <section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end" role="toolbar">
        <router-link :to="{ name: 'media'}" class="mdc-button top-bar__nav-button">
          <div class="mdc-button__ripple"></div>
          <span class="mdc-button__label">Musik library</span>
          <i class="material-icons mdc-button__icon" aria-hidden="true">library_music</i>
        </router-link>
      </section>
    </div>
  </header>
</template>

<script>
import { MDCTopAppBar } from "@material/top-app-bar";

export default {
  components: {},
  computed: {},
  data() {
    return {
      mdcAppBar: undefined
    };
  },
  mounted() {
    this.mdcAppBar = MDCTopAppBar.attachTo(this.$el);
  },
  beforeDestroy() {
    if (typeof this.mdcAppBar !== "undefined") {
      this.mdcAppBar.destroy();
    }
  }
};
</script>
<style lang="scss">
@use '@material/top-app-bar/mdc-top-app-bar';
@use '@material/button/mdc-button';
@use '@material/button';
@use '@material/theme';

.top-bar__nav-button {
  @include button.ink-color(theme.prop-value(on-primary));
}
</style>
