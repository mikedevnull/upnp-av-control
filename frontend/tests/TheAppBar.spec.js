import { mount, RouterLinkStub, enableAutoDestroy } from '@vue/test-utils'
import TheAppBar from '@/components/TheAppBar.vue'

enableAutoDestroy(afterEach)

describe('TheAppBar.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(TheAppBar, { stubs: { RouterLink: RouterLinkStub } });
    expect(wrapper.element).toMatchSnapshot();
  })
})
