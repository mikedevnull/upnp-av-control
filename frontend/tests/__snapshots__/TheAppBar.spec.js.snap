// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheAppBar.vue renders correctly 1`] = `
<header
  class="mdc-top-app-bar"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__title"
      >
        UPnP AV controller
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    >
      <a
        class="mdc-button top-bar__nav-button"
      >
        <div
          class="mdc-button__ripple"
        />
         
        <span
          class="mdc-button__label"
        >
          Musik library
        </span>
         
        <i
          aria-hidden="true"
          class="material-icons mdc-button__icon"
        >
          library_music
        </i>
      </a>
    </section>
  </div>
</header>
`;
